app/models/time_entry_activity.rb

```diff
@@ -42,20 +45,15 @@ def objects_count
   end
 
   def transfer_relations(to)
-    time_entries.update_all("activity_id = #{to.id}")
+    time_entries.update_all(activity_id: to.id)
   end
 
-  def activated_projects
-    scope = Project.all
-
-    scope = if active?
-              scope
-                .where.not(id: children.select(:project_id))
-            else
-              scope
-                .where('1=0')
-            end
+  def active_in_project?(project)
+    teap = time_entry_activities_projects.detect { |t| t.project_id == project.id }
```

Comment:
As the entries are eager loaded in the use case the method is currently employed in (https://github.com/opf/openproject/blob/feature/api_time_entries_forms/app/views/projects/form/_activities.html.erb#L34-L44) I deliberately wrote the method like this. But I now inserted a switch to also use sql based method in case they are not eager loaded.

Reply:
Ah right. Makes sense!